crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -482,4 +624,95 @@ mod tests {
             "git+https://github.com/flox/flox"
         );
     }
+
+    #[cfg(feature = "impure-unit-tests")] // disabled for offline builds, TODO fix tests to work with local repos
+    #[tokio::test]
+    async fn git_file_error_if_dirty() {
+        use std::fs;
+        use std::path::Path;
+        use std::str::FromStr;
+        env_logger::init();
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let repo_dir = _temp_dir_handle.path().join("repo");
+
+        fs::create_dir(&repo_dir).unwrap();
+        let repo = Git::init(&repo_dir, false).await.unwrap();
+
+        fs::write(repo_dir.join("flake.nix"), "{ outputs = _: {}; }").unwrap();
```

Comment:
For these tests it would be helpful if there was a comment or two explaining why you're doing what you're doing. In this case something like "create a file and stage it without committing so that the repo is dirty".